runtime/component/dnd-mode.js

```diff
@@ -0,0 +1,634 @@
+'use strict'
+
+var property = require('@yoda/property')
+var logger = require('logger')('dndmode')
+
+var FSMCode = {
+  Start: 0,
+  End: 254,
+  WaitAsync: 255,
+  CheckSwitchOn: 1,
+  CheckSwitchOff: 2,
+  CheckStatusOff: 3,
+  CheckStatusOn: 4,
+  DNDModeDisabled: 5,
+  CheckStatusOnX: 6,
+  CheckStatusOffX: 7,
+  CheckTimeSuccess: 8,
+  CheckTimeFailed: 9,
+  DNDModeDisabledX: 10,
+  CheckTimeSuccessX: 11,
+  CheckTimeFailedX: 12,
+  CheckActivityTrue: 13,
+  CheckActivityFalse: 14,
+  CheckActivityTrueX: 15,
+  CheckActivityFalseX: 16,
+  DNDModeEnabled: 17,
+  CheckAgain: 18
+}
+
+var SWITCH_KEY = 'dndmode.switch'
+var STATUS_KEY = 'dndmode.status'
+var WEEK_KEY = 'dndmode.weeksetting'
+var START_TIME_KEY = 'dndmode.starttime'
+var END_TIME_KEY = 'dndmode.endtime'
+var AWAKE_SWITCH_KEY = 'dndmode.awakeswitch'
+
+var FSM_READY = 0
+var FSM_RUNNING = 1
+var FSM_WAITING = 2
+var FSM_END = 3
+var FSM_ERROR = 4
+
+class DNDCommon {
+  constructor (life) {
+    this.life = life
```

Comment:
Lifetime API has been replaced by visibility API, checkout https://github.com/yodaos-project/yodart/pull/696 for details.

Reply:
> Lifetime API has been replaced by visibility API, checkout #696 for details.

OK.